core/src/main/kotlin/fr/sncf/osrd/stdcm/infra_exploration/InfraExplorer.kt

```diff
@@ -298,28 +304,31 @@ private class InfraExplorerImpl(
             if (!seenFirstBlock) {
                 nBlocksToSkip++
             } else {
+                val startPath = !pathStarted
+                val addRoute = block == routeBlocks.first() || startPath
+                val travelledPathBeginBlockOffset =
```

Comment:
If I understood correctly, it would clearer IMO to have explicit type:
```suggestion
                val travelledPathBegin: Offset<Block> =
```